### DISCUSSION TITLE
Syntax of function analyzing

### DISCUSSION THREAD

**Discussion by user18x** [ThunderCls/xAnalyzer#59] [Ideas]
It's possible to extend xanalyzer to see what commands (push...) are - as example - for GetMessageBox, GetDlgItemTextA etc.?
Like we can use "analyze selection" with xanalyzer.

Example:
C++ code from [here](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-messagebox)

```
int MessageBox(
  [in, optional] HWND    hWnd,
  [in, optional] LPCTSTR lpText,
  [in, optional] LPCTSTR lpCaption,
  [in]           UINT    uType
);
```
Now if i select a part of the code and right click -> xanalyzer -> analyse selection i see what belongs togehter.
But i don't see waht belongs together (pic1) in the GetMessageBox as assembler code.

pic1:
![GetMessageBox](https://user-images.githubusercontent.com/37771000/158060493-a45cbf5a-930a-4ef2-bccc-232efef71f28.jpg)

pic2
![GetMessageBox_2](https://user-images.githubusercontent.com/37771000/158060502-6d52c6d6-36f2-4e87-a84c-675223a67a1b.jpg)

Hope know you understand.

**Comment by ThunderCls** [comment]
Analysis won't work fine for calls with jumps and/or returns among arguments 
Read more at: https://github.com/ThunderCls/xAnalyzer#known-issues-and-limitations
